### DISCUSSION TITLE
How to asynchronously render a component only after fetching data from API?

### DISCUSSION THREAD

**Discussion by GorvGoyl** [vercel/next.js#23813] [Help] [answered]
I've 2 doubts:

1. Can I load a component asynchronously in a page? This component should fetch the data from api and then render. Rest of the page *shouldn't wait* for its response and render anyway. Think of this component as an `iFrame`.

2. Also, I'm thinking of calling API and fetching data inside `useEffect` hook of this component. Is this the right approach?

**Comment by akellbl4** [comment]
Hi!
1. Yes, you can load components asynchronously. Use `next/dynamic`. [Here are docs](https://nextjs.org/docs/advanced-features/dynamic-import)
2. If you want to render that component only on the client side that's totally fine.

**Comment by GorvGoyl** [comment, marked as answer]
Figured out I could easily do it by calling async api inside `useEffect` hook and maintain a `state` var which would render the component once the call is finished.
